doc/RELEASE_PROCESS.md

```diff
@@ -0,0 +1,25 @@
+# How to make a keyvi release (e.g., vX.Y.Z)
+
+## Create a release branch
+Create a release branch called `release-X.Y.Z`
+
```

Comment:
It would be clearer to me to mention branches more top level:

On the `release branch`:
...

On the `master branch`:
...